test/PowerShellEditorServices.Test/Refactoring/RefactorVariableTests.cs

```diff
@@ -0,0 +1,299 @@
+
+            string modifiedcontent = TestRenaming(scriptFile, request);
+
+            Assert.Equal(expectedContent.Contents, modifiedcontent);
+
+        }
+        [Fact]
+        public void VariableScriptWithParamBlock()
+        {
+            RenameSymbolParams request = RenameVariableData.VariableScriptWithParamBlock;
+            ScriptFile scriptFile = GetTestScript(request.FileName);
+            ScriptFile expectedContent = GetTestScript(request.FileName.Substring(0, request.FileName.Length - 4) + "Renamed.ps1");
+
+            string modifiedcontent = TestRenaming(scriptFile, request);
+
+            Assert.Equal(expectedContent.Contents, modifiedcontent);
+
+        }
+        [Fact]
+        public void VariableNonParam()
+        {
+            RenameSymbolParams request = RenameVariableData.VariableNonParam;
+            ScriptFile scriptFile = GetTestScript(request.FileName);
+            ScriptFile expectedContent = GetTestScript(request.FileName.Substring(0, request.FileName.Length - 4) + "Renamed.ps1");
+
+            string modifiedcontent = TestRenaming(scriptFile, request);
+
+            Assert.Equal(expectedContent.Contents, modifiedcontent);
+
+        }
+        [Fact]
+        public void VariableParameterCommandWithSameName()
+        {
+            RenameSymbolParams request = RenameVariableData.VariableParameterCommandWithSameName;
+            ScriptFile scriptFile = GetTestScript(request.FileName);
+            ScriptFile expectedContent = GetTestScript(request.FileName.Substring(0, request.FileName.Length - 4) + "Renamed.ps1");
+
+            string modifiedcontent = TestRenaming(scriptFile, request);
+
+            Assert.Equal(expectedContent.Contents, modifiedcontent);
+        }
+        [Fact]
+        public void VarableCommandParameterSplattedFromCommandAst()
+        {
+            RenameSymbolParams request = RenameVariableData.VariableCommandParameterSplattedFromCommandAst;
+            ScriptFile scriptFile = GetTestScript(request.FileName);
+            ScriptFile expectedContent = GetTestScript(request.FileName.Substring(0, request.FileName.Length - 4) + "Renamed.ps1");
+
+            string modifiedcontent = TestRenaming(scriptFile, request);
+
+            Assert.Equal(expectedContent.Contents, modifiedcontent);
+        }
+        [Fact]
+        public void VarableCommandParameterSplattedFromSplat()
+        {
+            RenameSymbolParams request = RenameVariableData.VariableCommandParameterSplattedFromSplat;
+            ScriptFile scriptFile = GetTestScript(request.FileName);
+            ScriptFile expectedContent = GetTestScript(request.FileName.Substring(0, request.FileName.Length - 4) + "Renamed.ps1");
+
+            string modifiedcontent = TestRenaming(scriptFile, request);
+
+            Assert.Equal(expectedContent.Contents, modifiedcontent);
+        }
+        [Fact]
+        public void VariableInForeachDuplicateAssignment()
```

Comment:
Something in this test is vfailing on MacOS
https://github.com/PowerShell/PowerShellEditorServices/actions/runs/9378437855/job/25846949282?pr=2152

Reply:
I was running my tests in .NET 7, my bad. I get the same error on windows with .NET 8, its to do with how the scriptast is getting generated. it was only creating essentially the first line of the scriptAST ... adding a new line at the start to the test file gets it to generate it correctly. to me this seems like a bug in powershell or the workspace.GetFile method

Below is the Debug without the newline at the start of the file
![image](https://github.com/PowerShell/PowerShellEditorServices/assets/3089087/262be2b0-4180-4723-ab08-455b352ec747)

This is with the newline added at the start
![image](https://github.com/PowerShell/PowerShellEditorServices/assets/3089087/f7ae5317-98f4-4377-9089-4566f1c294e4)


I'm using the below method to generate that scriptFile object
```Csharp
private ScriptFile GetTestScript(string fileName) => workspace.GetFile(TestUtilities.GetSharedPath(Path.Combine("Refactoring", "Variables", fileName)));
```